### PR TITLE
Cleanup and document _plot_args()

### PR DESCRIPTION
**Pull request by timhoffm**

### CODE REVIEW

**Review by jklymak — APPROVED** [review]

**Comment by timhoffm** [comment]
Test failure is unredated (segfault in PyQt).

File: lib/matplotlib/axes/_base.py [hunk 1]
```diff
@@ -417,16 +458,16 @@ def _plot_args(self, tup, kwargs, return_kwargs=False):
             raise ValueError("x, y, and format string must not be None")
```
Comment (QuLogic) [inline lib/matplotlib/axes/_base.py:458]: This would never have checked whether `fmt` is None, because of the first if, is it not?

Comment (timhoffm) [reply lib/matplotlib/axes/_base.py:458]: For `(x, y, None)`, you ended up with the "third arg must be a format string" error. But for `(y, None)` this check triggers.

Comment (QuLogic) [reply lib/matplotlib/axes/_base.py:458]: Ah yes, because the length checks are different.

**Review by QuLogic — APPROVED** [review]

**Pull request merged**